### ISSUE TITLE
 Style/MethodCallWithArgsParentheses false positive with pattern matching

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop-hq/rubocop#8904]
## Expected behavior

I would expect the following code to not throw a *Style/MethodCallWithArgsParentheses* error because the parentheses are necessary and cannot be omitted in this situation:

``` ruby
case settings
  in build_minimum: true, **remainder then build(:build_minimum, settings)
  in build:, **remainder then build(:build_maximum, options)
end
```

## Actual behavior

```
rubocop --debug example.rb
For /Users/bkuhlmann/Downloads: configuration from /Users/bkuhlmann/.config/rubocop/config.yml
configuration from /Users/bkuhlmann/.config/rubocop/.rubocop-https---raw-githubusercontent-com-bkuhlmann-code-quality-master-configurations-rubocop-ruby-yml
Default configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-0.93.1/config/default.yml
configuration from /Users/bkuhlmann/.config/rubocop/.rubocop-https---raw-githubusercontent-com-bkuhlmann-code-quality-master-configurations-rubocop-rake-yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-rake-0.5.1/config/default.yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-rake-0.5.1/config/default.yml
configuration from /Users/bkuhlmann/.config/rubocop/.rubocop-https---raw-githubusercontent-com-bkuhlmann-code-quality-master-configurations-rubocop-performance-yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-performance-1.8.1/config/default.yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-performance-1.8.1/config/default.yml
configuration from /Users/bkuhlmann/.config/rubocop/.rubocop-https---raw-githubusercontent-com-bkuhlmann-code-quality-master-configurations-rubocop-rspec-yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-rspec-1.43.2/config/default.yml
configuration from /Users/bkuhlmann/.gem/ruby/2.7.2/gems/rubocop-rspec-1.43.2/config/default.yml
Inspecting 1 file
Scanning /Users/bkuhlmann/Downloads/example.rb
C

Offenses:

example.rb:17:49: C: Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
  in build_minimum: true, **remainder then build(:build_minimum, settings)
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^
example.rb:18:36: C: Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
  in build:, **remainder then build(:build_maximum, options)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable
Finished in 0.6471289999317378 seconds
```

## Steps to reproduce the problem

You can reproduce this problem by adding the following configuration to your `.rubocop.yml`:

``` yaml
Style/MethodCallWithArgsParentheses:
  Enabled: true
  EnforcedStyle: omit_parentheses
  AllowParenthesesInChaining: true
  AllowParenthesesInMultilineCall: true
  IgnoreMacros: false
```

Next, save the following as `example.rb`

``` ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile true do
  source "https://rubygems.org"
  gem "rubocop", "~> 0.93"
end

def build key, settings
  puts "#{key}, #{settings}"
end

settings = {build_minimum: true, build: "test"}

case settings
  in build_minimum: true, **remainder then build(:build_minimum, settings)
  in build:, **remainder then build(:build_maximum, settings)
end
```

Finally, run the following to reproduce the error:

``` bash
rubocop example.rb
```

## RuboCop version

```
rubocop -V
0.93.1 (using Parser 2.7.2.0, rubocop-ast 0.8.0, running on ruby 2.7.2 x86_64-darwin19)
```

**Comment by marcandre** [comment]
Thanks for the bug report.

This is not really related to pattern matching per say.

The cop should allow (probably) allow method calls when there's a local variable conflict (like in your example):

```ruby
build = :something
build(42) # method call
```

You could not use parenthesis by calling `self.build` in your example. It would have the benefit of being clear.

**Comment by bkuhlmann** [comment]
:bow: @marcandre. You are correct. Using `self.build` works but then leads to a `Style/RedundantSelf` error. Renaming the private `build` method to something other than `build` ends up resolving both errors and yielding zero Rubocop violations.

What tripped me up was the original `Style/MethodCallWithArgsParentheses` error was misleading and I lost focus of what the root cause was.

I don't know of how to make the error message better, though? I could see `Style/MethodCallWithArgsParentheses` attempting to print out a better suggestion with the error but seems like too much responsibility?

I can close this issue if no one thinks there is further action here?

**Comment by bkuhlmann** [comment]
ℹ️ I'm closing this issue as I'm not sure what the best course of action is. Despite the error message being slightly confusing, upon refactoring the code as mentioned in the previous comment, the error is resolved. Probably best to see if more people stumble upon this before pursuing further.
